packages/workshop-app/app/routes/launch-editor.tsx

```diff
@@ -69,8 +71,24 @@ export function LaunchEditor({
 } & (
 	| { file: string; appFile?: never; appName?: never }
 	| { file?: never; appFile: string; appName: string }
+	| { file?: never; appFile: string[]; appName: string }
 )) {
 	const fetcher = useFetcher<typeof action>()
+
+	if (fetcher.state === 'loading') {
+		const error = fetcher.data?.status === 'error' ? fetcher.data.error : ''
+		if (error) {
+			const detail: EventDetail = {
+				title: 'Launch Editor Error',
+				content: error,
+			}
+			const event = new CustomEvent<EventDetail>('kcdshop-error', { detail })
+			document.dispatchEvent(event)
+		}
+		const submitted = new CustomEvent('kcdshop-launchEditor-submitted')
+		document.dispatchEvent(submitted)
```

Comment:
Instead of constructing these events here leading to some indirection, could we make a function that inside the same file where the listeners are?

Reply:
working on it

Reply:
by the way, this is not always possible, when you have one dispatcher and many listeners in more than one file